### PR TITLE
feat: revert the revert of "chore: remove unneeded market fetch (#1808)"

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Reverts the revert of b7ca8078b93f0d24d2d75594bd1e9bd73f4eeb09

Brings back the XYZ on Osmosis assets, which were showing an infinite Skeleton loading as the request to get the Market data was never fetched for these.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

https://github.com/shapeshift/web/pull/1808

## Risk

This touches `<AppContext />` so as always with this, we should be very cautious, many things can break.
Effectively, there's two layers of `useEffect`s , each with their relevant protection, so the only risk is overfetching market data in `/assets/:chainId/:assetSubId` routes, which itself is mitigated by the fact that there's caching involved.

Tl;dr big risks, low effective risks

## Testing

- Go to Atom on Osmosis (or any XYZ on Osmosis asset)
- The asset page should load
- There should be no regression on other asset pages

## Screenshots (if applicable)

image

### CODE REVIEW

**Review by 0xdef1cafe — APPROVED** [review]

**Review by elmutt — APPROVED** [review]

**Review by 0xdef1cafe — APPROVED** [review]

**Review by 0xApotheosis — APPROVED** [review]
🍭
